doc/barman.md

```diff
@@ -0,0 +1,136 @@
+
+[Barman](http://www.pgbarman.org) (Backup and Recovery Manager) is an open-source administration tool for disaster recovery of PostgreSQL servers.
+
+This example shows how to integrate Barman [Barman](http://www.pgbarman.org) with Stolon. Therefore, by integrating Barman with Stolon,
+you can dramatically reduce RTO(Recovery Time Objective) and RPO(Recovery Point Objective) to nearly zero values.
+
+### Introduction
+
+Offers two options for backup and archiving. Traditionally, Barman has supported standard WAL file shipping through
+PostgreSQL’s `archive_command` (usually via `rsync/SSH`). With this method, WAL files are archived only when PostgreSQL switches to a new WAL file.
+To keep it simple, this normally happens every 16 MB worth of data changes. Barman also introduces streaming of WAL files for PostgreSQL 9.2 or later,
+as an additional method for transactional log archiving through `pg_receivexlog`. WAL streaming is able to reduce the risk of data loss,
+bringing RPO down to nearly zero values. In addition, Barman introduces support for replication slots with PostgreSQL servers 9.4 or later, therefore allowing WAL streaming-only configurations.
+Moreover, you can now add Barman as a synchronous WAL receiver in your PostgreSQL 9.5 (or later) cluster and achieve zero data loss (RPO=0).
+Thus, the traditional archiving of WAL logs can lead to data loss in the event of an accident.
+
+### Possible solution for integration of Stolon with Barman
+
+![Stolon architecture with Barman](stolon_and_barman.png)
+
+For integration of  Stolon with Barman, you can use scenarios for backup via streaming protocol.
+
+In this scenario, you will need to configure:
+
+- A standard connection to PostgreSQL for management, coordination, and monitoring purposes,
+- A streaming replication connection that will be used by both pg_basebackup (for base backup operations) and pg_receivexlog (for WAL streaming)
+This setup, in Barman’s terminology, is known as streaming-only setup, as it does not require any SSH connection for backup and archiving operations.
+This is particularly suitable and extremely practical for Docker environments.
+
+But for restoring Stolon Cluster from Barman backup you should configure SSH connections between PostgreSQL and Barman nodes.
+
+#### Base backups
+
+##### Barman configurations
+
+Example for Streaming-Only backups (/etc/barman/stolon.conf)
+
+```
+[stolon]
+description =  "Stolon Backup (Streaming-Only)"
+conninfo = host=primaryhost user=barman dbname=postgres port=5432
+streaming_conninfo = host=primaryhost user=replica port=5432
+backup_method = postgres
+retention_policy = RECOVERY WINDOW OF 7 DAYS
+wal_retention_policy = main
+retention_policy_mode = auto
+minimum_redundancy=2
+streaming_backup_name = barman_streaming_backup
+streaming_archiver = on
+slot_name = add_slot
+streaming_archiver_name = barman_receive_wal
+```
+
+##### Creating replication slot for Barman
+
+```
+barman receive-wal --create-slot stolon
```

Comment:
I'll name the slot to something like `barman`

Reply:
This will also start a `receive-wal` process not just create a repl slot. What about removing all this part (`##### Creating replication slot for Barman`) and just add some notes on how to create repl slots (using the `--create-slot` or the `additionalMasterReplicationSlots` cluster spec option?) below (in `#### Starting replication from Stolon to Barman`)?